### ISSUE TITLE
The SqLite client database schema is not updated by the interceptor OnConflictingSetup

### ISSUE THREAD

**Issue by ricardoamm** [Mimetis/Dotmim.Sync#1085]
Hi,

We needed to add a new column on the server side, and performed the in place migration on the server side.
According to the documentation, we implement the interceptor OnConflictingSetup, but the local database schema was not updated.
How can this be done automatically, without the need to update the schema manually?

Best regards

**Comment by Mimetis** [comment]
The `OnConflictingSetup` will not update the local schema.

[As mentioned here](https://dotmimsync.readthedocs.io/Provision.html?highlight=OnConflictingSetup#client-side), you’re still responsible of migrating your schema in your local database.

Once it's done (removing or adding columns or table) you need to reprovision the client side to be sure **DMS** metadata, triggers and stored procedures are generated correctly with the new added columns / tables.

You can do it by calling the `ProvisionAsync` method or you can let **DMS** handles it using the `OnConflictingSetup` interceptor

**Comment by ricardoamm** [comment]
Thanks for your quick reply.

I removed a not null column in the server database and called ProvisionAsync. 
On the client side, before sync, I updated the SqlLite Database schema and then called ProvisionAsync, but I'm having the following error:

11:06:43] Error: [InternalApplyChangesAsync]..[InternalApplyChangesAsync]..[InternalApplyTableChangesAsync]..Specified argument was out of the range of valid values. (Parameter 'Can't make a query on primary keys since number of primary keys columns in criteria is not matching the number of primary keys columns in this table'
